workflow-service/src/main/java/com/redhat/parodos/workflow/execution/repository/WorkFlowRepository.java

```diff
@@ -44,7 +44,7 @@ WorkFlowExecution findFirstByMasterWorkFlowExecutionAndWorkFlowDefinitionId(
 	@Query("SELECT w FROM workflow_execution w WHERE w.status IN :statuses")
 	List<WorkFlowExecution> findByStatusIn(@Param("statuses") List<WorkFlowStatus> statuses);
 
-	@Query("SELECT w FROM workflow_execution w WHERE w.masterWorkFlowExecution.id = :masterWorkflowId and EXISTS (SELECT f.type FROM workflow_definition f WHERE f.id = w.workFlowDefinitionId AND f.type = com.redhat.parodos.workflow.enums.WorkFlowType.CHECKER)")
-	List<WorkFlowExecution> findCheckers(@Param("masterWorkflowId") UUID masterWorkflowId);
+	@Query("SELECT w FROM workflow_execution w WHERE w.status = com.redhat.parodos.workflow.enums.WorkFlowStatus.FAILED and w.masterWorkFlowExecution.id = :masterWorkflowId and EXISTS (SELECT f.type FROM workflow_definition f WHERE f.id = w.workFlowDefinitionId AND f.type = com.redhat.parodos.workflow.enums.WorkFlowType.CHECKER)")
```

Comment:
Just a question, this `:masterWorkflowId` is protected with sqlinjections?

Reply:
yes,   this is jpql feature that parameters can be definied with prefix ':'

Reply:
masterWorkflowId is UUID not string